File: 3Bimestre/Backend/CTD-Backend/src/TFI/Services/PatientService.java
package TFI.Services;

import TFI.DAO.DAO;
import TFI.Models.Patient;

import java.util.List;

public class PatientService extends UserService {
    private DAO<Patient> DAO;

    public PatientService(DAO<Patient> DAO) {
        this.DAO = DAO;
    }

    public Patient createPatient(Patient patient) {
        int id = DAO.create(patient);
        patient.setId(id);
        return patient;
    }

    public Patient updatePatient(int id, Patient patient) {
        return DAO.update(id, patient);
    }

    public void deletePatient(int id) {
        DAO.delete(id);
    }

    public Patient getPatient(int id) {
        return DAO.get(id);
    }

    public List<Patient> listPatients() {
        return DAO.list();
    }
}

File: 3Bimestre/Backend/TFI/src/main/frontend/src/services/patientService.ts
import { HTTPMethods, IPatient, IPatientList } from '../types';
import { fetchAPI } from './fetchAPI';

export const listPatients = async (): Promise<IPatientList[]> => {
  return await fetchAPI(HTTPMethods.Get, 'patients/');
};

export const getPatient = async (id: string): Promise<IPatient> => {
  return await fetchAPI(HTTPMethods.Get, 'patients/' + id);
};

export const createPatient = async (patient: IPatient): Promise<IPatient> => {
  return await fetchAPI(HTTPMethods.Post, 'patients/', patient);
};

export const updatePatient = async (id: string, updatedPatient: IPatient): Promise<IPatient> => {
  return await fetchAPI(HTTPMethods.Put, 'patients/' + id, updatedPatient);
};

export const deletePatient = async (id: string): Promise<string> => {
  return await fetchAPI(HTTPMethods.Delete, 'patients/' + id);
};
